wolfhsm/wh_comm.h

```diff
@@ -97,17 +145,17 @@ typedef struct {
  * request sequence number and provide a buffer for at least 1 packet.
  */
 typedef struct {
+    uint64_t packet[WH_COMM_MTU_8];
     void* transport_context;
     const whTransportClientCb* transport_cb;
-    uint16_t reqid;
-    uint16_t seq;
-    uint16_t size;
-    uint8_t packet[WH_COMM_MTU];
     whCommHeader* hdr;
     uint8_t* data;
     uint32_t client_id;
     uint32_t server_id;
     int initialized;
+    uint16_t reqid;
+    uint16_t seq;
+    uint16_t size;
 } whCommClient;
```

Comment:
I think we should combine the client and server comm types and functions since they're so similar, really seems like a waste of code size

Reply:
Good idea.  At the end of the day, the callbacks are distinct, but there's probably a common struct and accessors that we can combine to handle the buffering and comm states, even if there are extra items that aren't used in the server side.  That would simplify the code and reduce cut-n-paste errors.  The send and recv funcs are nearly identical, except for that actual callback to the transport.  Lemme take a peek at this, as I was wanting to split the comms into wh_comm.h, wh_commclient.h, and wh_commserver.h, as well as the C files to improve isolation.  I'll add this to the backlog.